main.cpp

```diff
@@ -102,18 +103,38 @@ void uthread_create(void *(start_routine)(void *), void* arg){
     tcb->arg = arg;
 
     // Allocate the stack.
-    auto* stack = (char*)malloc(STACK_SIZE);
+    auto* stack = (char *)malloc(STACK_SIZE);
```

Comment:
Just out of curiosity, why does it work to have the stack as an array of chars? It seems like it would need to be of type address_t.